pallets/gear/src/tests.rs

```diff
@@ -14697,6 +14697,126 @@ fn test_gas_info_of_terminated_program() {
     })
 }
 
+#[test]
+fn test_handle_signal_wait_old() {
```

Comment:
merge initial version with this pr, please
in several - adapt it to constructor